hs_core/task_utils.py

```diff
@@ -0,0 +1,30 @@
+from celery.task.control import inspect
+
+
+def get_resource_bag_task(res_id):
```

Comment:
I think we need a test around this for each of the queues.  Do you have real examples of what reserved and scheduled looks like?  Is it the same format as active?  If it is, then you can put the parsing into it's own method and just call it for each queue instead of duplicating the code.

Reply:
Good point on removing code duplication. I have done extensive testing using django shell and also read through the documentation here: http://docs.celeryproject.org/en/latest/userguide/workers.html?highlight=revoke#inspecting-workers, and can assert that active and reserved jobs have the same format but scheduled jobs have a bit different format. I have tried to encapsulate the difference in an internal method to get the task id. Please take a quick look again to see if you see any issues. I did step through each line of the code to make sure the routine works for both active and scheduled jobs. I cannot get to a scenario of non-empty reserved job queue, so I cannot test reserved job scenario, but based on the documentation, reserved job scenario has the same format as active job, so I assume it should work for reserved job queue as well. In terms of creating a test around this, I am not sure how to write a useful test for this. Let me know if you have specific ideas, though.